src/extension/crashreporter.ts

```diff
@@ -1,39 +1,21 @@
-import * as Sentry from "@sentry/node";
-import { v4 as uuidv4 } from "uuid";
-import {
-    workspace,
-} from "vscode";
+import TelemetryReporter from "vscode-extension-telemetry";
 
 export class CrashReporter {
-    private enableCrashReporting: boolean | undefined;
+    private reporter: TelemetryReporter;
 
     constructor() {
-        const telemetry = workspace.getConfiguration("telemetry");
-        this.enableCrashReporting = telemetry.get("enableCrashReporter");
-
-        if (this.enableCrashReporting) {
-            // Leaving default integrations on captures crashes from other extension hosts
-            // Turning this off fixes that issue and still allows us to capture errors manually
-            Sentry.init({
-                defaultIntegrations: false,
-                dsn: "https://dfd1a0d586284b6b8710feef8a2928b3@o412074.ingest.sentry.io/5288283",
-                release: "vscode-coverage-gutters@2.8.1",
-            });
-            Sentry.configureScope(function(scope) {
-                // Generate a random string for this session
-                // Note: for privacy reason, we cannot fingerprint across sessions
-                scope.setUser({id: uuidv4()});
-            });
-        }
+        this.reporter = new TelemetryReporter(
+            "vscode-coverage-gutters",
+            "2.9.0",
```

Comment:
[F] I think that you can get this from `ExtensionContext.extension.packageJSON.version` (In case you don't want to manually enter this each time you bump the version).

Reply:
I also just realised this is an unsolicited comment, sorry if it is unuseful

Reply:
@mattseddon I will never say no to feedback! Thanks for the suggestion, looks better to me and saves me having to iterate this all the time 🤔.

Reply:
@mattseddon I did find that context doesn't have access to this but it looks like it is more standard to use `getExtension()`.
https://github.com/microsoft/vscode-gather/blob/4aca410f888efc0c56e8f45aacda2b8bd0ef3b1b/src/telemetry.ts#L94-L96

Reply:
It seems serendipitous that you are doing this now as I am also working on putting analytics in this week. 

I have seen that generally all of the Microsoft repositories copy + paste that same file for dealing with the vscode-extension-telemetry instance (vscode-edge-devtools, vscode-jupyter & vscode-python were the first 3 I checked).

My initial thought was that I was going to go in a different direction and have a single instance of the reporter injected into all of the classes that I needed (hence I thought it would be easy to get the version from `activate(context: ExtensionContext)`). 

After some more thought / digging it appears that I won't actually get that much from doing that so this I'll be going the same route as "everyone" else.

Sorry for the dud suggestion.

Reply:
@mattseddon interesting, there are some advantages to having one instance though as it is easier to call dispose on later but as you mentioned it seems all of Microsoft's implementations revolve around creating a new reporter each time...

Reply:
From what I've seen it's actually the same "global" instance for each call of `getTelemetryReporter`. they'll only ever instantiate one unless they clear it:

```
let telemetryReporter: TelemetryReporter | undefined;
function getTelemetryReporter() {
    if (!isTestExecution() && telemetryReporter) {
        return telemetryReporter;
    }
    const extensionId = JVSC_EXTENSION_ID;
    // eslint-disable-next-line @typescript-eslint/no-require-imports
    const extensions = (require('vscode') as typeof import('vscode')).extensions;
    const extension = extensions.getExtension(extensionId)!;
    const extensionVersion = extension.packageJSON.version;

    // eslint-disable-next-line @typescript-eslint/no-require-imports
    const reporter = require('vscode-extension-telemetry').default as typeof TelemetryReporter;
    return (telemetryReporter = new reporter(extensionId, extensionVersion, AppinsightsKey, true));
}
```

We use `@hediet/std/disposable` (because it gives us "hot reload" for dev) which means that we just have to "track" that global reporter somewhere and it will get cleaned up when the parent class gets disposed.

something like:
```
this.dispose.track(getTelementryReporter())
```
in the main `Extension` class.

Reply:
@mattseddon ahhh gotcha, great explanation! I see what you mean now, by the singleton they got going on there 🤔, crafty! Definitely seems like the way to go for dispose as well given it ultimately just cleans up the config listeners and sends any pending events.
https://github.com/microsoft/vscode-extension-telemetry/blob/main/src/common/baseTelemetryReporter.ts#L319-L325